cobalt/dom/source_buffer_metrics.cc

```diff
@@ -121,21 +121,33 @@ void SourceBufferMetrics::EndTracking(SourceBufferMetricsAction action,
 #endif  // !defined(COBALT_BUILD_TYPE_GOLD)
 }
 
-void SourceBufferMetrics::RecordTelemetry(
-    SourceBufferMetricsAction action, const base::TimeDelta& action_duration) {
+void SourceBufferMetrics::RecordUMATelemetry(
+    SourceBufferMetricsAction action, const base::TimeDelta& action_duration,
+    size_t size_appended) {
   switch (action) {
     case SourceBufferMetricsAction::PREPARE_APPEND:
       UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
           "Cobalt.Media.SourceBuffer.PrepareAppend.Timing", action_duration,
           base::TimeDelta::FromMicroseconds(1),
           base::TimeDelta::FromMilliseconds(25), 50);
       break;
-    case SourceBufferMetricsAction::APPEND_BUFFER:
+    case SourceBufferMetricsAction::APPEND_BUFFER: {
       UMA_HISTOGRAM_CUSTOM_MICROSECONDS_TIMES(
           "Cobalt.Media.SourceBuffer.AppendBuffer.Timing", action_duration,
           base::TimeDelta::FromMicroseconds(1),
           base::TimeDelta::FromMilliseconds(25), 50);
+
+      UMA_HISTOGRAM_COUNTS_10M(
```

Comment:
Can you look at the UMA_HISTOGRAM_MEMORY* set of macros and verify you don't want to use those instead? https://github.com/youtube/cobalt/blob/29e0d650bb4b88431aed9f4afcb983b41dd19ce3/base/metrics/histogram_macros.h#L329

Reply:
These look to be for sizes much larger than this data is expected to be.

`UMA_HISTOGRAM_MEMORY_KB` says it's for memory stats of size 500,000kb to 500M.

From local testing, I expect the sizes here to be anywhere between a few hundred bytes to ~10000kb at the maximum.